query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryTimeRange.java

```diff
@@ -0,0 +1,15 @@
+package org.hypertrace.core.query.service;
+
+import java.time.Duration;
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
```

Comment:
Please always use `@Value` over getting and all args constructor. Value does a number of other things, making the classes final, generate reasonable equal/tostring/hashcode etc.